source/ota.c

```diff
@@ -1262,45 +1155,34 @@ static DocParseErr_t decodeAndStoreKey( char * pValueInJson,
                                         void * pParamAdd )
 {
     DocParseErr_t err = DocParseErrNone;
+    size_t actualLen = 0;
+    Base64Status_t base64Status = 0;
+    Sig256_t ** pSig256 = pParamAdd;
 
-    /* Allocate space for and decode the base64 signature. */
-    void * pSignature = otaAgent.pOtaInterface->os.mem.malloc( sizeof( Sig256_t ) );
+    /* pSig256 should point to pSignature in OtaFileContext_t, which is statically allocated. */
+    assert( *pSig256 != NULL );
 
-    if( pSignature != NULL )
-    {
-        size_t actualLen = 0;
-        int base64Status = 0;
-        Sig256_t ** pSig256 = pParamAdd;
-
-        *pSig256 = pSignature;
-
-        base64Status = base64Decode( ( *pSig256 )->data,
-                                     sizeof( ( *pSig256 )->data ),
-                                     &actualLen,
-                                     ( const uint8_t * ) pValueInJson,
-                                     valueLength );
+    base64Status = base64Decode( ( *pSig256 )->data,
+                                    sizeof( ( *pSig256 )->data ),
+                                    &actualLen,
+                                    ( const uint8_t * ) pValueInJson,
+                                    valueLength );
 
-        if( base64Status != 0 )
-        {
-            /* Stop processing on error. */
-            LogError( ( "Failed to decode Base64 data: "
-                        "base64Decode returned error: "
-                        "error=%d",
-                        base64Status ) );
-            err = DocParseErrBase64Decode;
-        }
-        else
-        {
-            ( *pSig256 )->size = ( uint16_t ) actualLen;
-            LogInfo( ( "Extracted parameter [ %s: %.32s... ]",
-                       OTA_JsonFileSignatureKey,
-                       pValueInJson ) );
-        }
+    if( base64Status != Base64Success )
+    {
+        /* Stop processing on error. */
+        LogError( ( "Failed to decode Base64 data: "
+                    "base64Decode returned error: "
+                    "error=%d",
+                    base64Status ) );
+        err = DocParseErrBase64Decode;
     }
     else
     {
-        /* We failed to allocate needed memory. Everything will be freed below upon failure. */
-        err = DocParseErrOutOfMemory;
+        ( *pSig256 )->size = ( uint16_t ) actualLen;
```

Comment:
Is this case not considered an error? Also, if actualLen is always supposed to be zero here, then we may just want to assign this to zero directly. This way it's more explicit that this is the behavior during an error.

Reply:
What do you mean? This is the success path right? `base64Status == Base64Success` in this else branch.

Reply:
You're right, I misread the condition